internal/credentials/kubernetes/ecr/managed_iam_credential.go

```diff
@@ -19,7 +19,7 @@ import (
 	"github.com/akuity/kargo/internal/logging"
 )
 
-type podIdentityCredentialHelper struct {
+type managedIAMCredentialHelper struct {
```

Comment:
"managed IAM" sounds redundant to me. (__Managed__ Identity and Access __Management__)

I think `managedIdentityCredentialHelper` seems like it accurately covers both Pod Identity and IRSA.